Karatsuba_Algorithm/Karatsuba_Algorithm.go

```diff
@@ -0,0 +1,124 @@
+// Karatsuba Algorithm In GO
+
+package main
+
+// Importing Necessary Packages
+import 
+(
+    "fmt"
+    "math"
+)
+
+// Input Function To Get The Digits
+func getDigits(number int64) uint {
+
+    var ans uint
+
+    if number == 0 {
+        return 1
+    }
+
+    if number < 0 {
+        number = -number
+    }
+
+    for number > 0 {
+        ans++
+        number = number / 10
+    }
+
+    return ans
+}
+
+func getHighAndLowDigits(num int64, digits uint) (int64, int64) {
+
+    divisor := int64(math.Pow(10, float64(digits)))
+
+    if num >= divisor {
+        return num / divisor, num % divisor
+    } 
+    else {
+        return 0, num
+    }
+}
+
+// Karatsuba Algorithm Function
+func karatsuba(x int64, y int64) int64 {
+
+    var max_digits uint
+    positive := true
+
+    if x == 0 || y == 0 {
+        return 0
+    }
+
+    if (x > 0 && y < 0) || (x < 0 && y > 0) {
+        positive = false
+    }
+
+    if x < 0 {
+        x = -x
+    }
+
+    if y < 0 {
+        y = -y
+    }
+
+    if x < 10 || y < 10 {
+        return x * y
+    }
+
+    x_digits := getDigits(x)
+    y_digits := getDigits(y)
+
+    if x_digits >= y_digits {
+        max_digits = x_digits / 2
+    } 
+    else {
+        max_digits = y_digits / 2
+    }
+
+    x_high, x_low := getHighAndLowDigits(x, max_digits)
+    y_high, y_low := getHighAndLowDigits(y, max_digits)
+
+    z0 := karatsuba(x_low, y_low)
+    z1 := karatsuba((x_low + x_high), (y_low + y_high))
+    z2 := karatsuba(x_high, y_high)
+
+    if positive {
+        return (z2 * int64(math.Pow(10, float64(2*max_digits)))) + (z1-z2-z0)*int64(math.Pow(10, float64(max_digits))) + z0
```

Comment:
Please split this line of code across two lines. It is too big.

Reply:
Also add space around operators in this (z1-z2-z0)*int64(math.Pow(10, float64(max_digits)))